File: api/app/models/despesa.ts
import { DateTime } from 'luxon'
import { BaseModel, column, belongsTo } from '@adonisjs/lucid/orm'
import type { BelongsTo } from '@adonisjs/lucid/types/relations'
import Parlamentar from './parlamentar.js'

export default class Despesa extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare parlamentarId: number

  @column()
  declare ano: number

  @column()
  declare mes: number | null

  @column()
  declare tipoDespesa: string | null

  @column()
  declare fornecedor: string | null

  @column()
  declare cnpjCpf: string | null

  @column()
  declare valorDocumento: number

  @column()
  declare valorLiquido: number

  @column()
  declare numeroDocumento: string | null

  @column()
  declare urlDocumento: string | null

  @column.date()
  declare dataEmissao: DateTime | null

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime

  @belongsTo(() => Parlamentar)
  declare parlamentar: BelongsTo<typeof Parlamentar>
}

File: api-dotnet/MonitorParlamentar.Domain/Entities/Despesa.cs
namespace MonitorParlamentar.Domain.Entities;

public class Despesa
{
    public int Id { get; set; }
    public int ParlamentarId { get; set; }
    public int Ano { get; set; }
    public int? Mes { get; set; }
    public string? TipoDespesa { get; set; }
    public string? Fornecedor { get; set; }
    public string? CnpjCpf { get; set; }
    public decimal ValorDocumento { get; set; }
    public decimal ValorLiquido { get; set; }
    public string? NumeroDocumento { get; set; }
    public string? UrlDocumento { get; set; }
    public DateOnly? DataEmissao { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
